woocommerce/woocommerce-android
# Move secrets other than .properties outside of the repository

Counter part to https://github.com/wordpress-mobile/WordPress-Android/pull/14995.

Moves the low-hanging fruits in terms of secrets outside of the repository:

- Fastlane's `google-upload-cerdentials.json`
- The debug keystore

## To test

Checkout this branch and run `bundle exec fastlane run configure_apply`, then `ls ~/.configure/woocommerce-android/secrets`, you should see:

```
➜ ls ~/.configure/woocommerce-android/secrets

debug_a8c.keystore             google-upload-credentials.json
```

Also ensure that `debugStoreFile` in the local `gradle.properties` has been updated to the new path from `~/.configure/woocommerce-android/secrets/`.

**Individual tests**

`debug_a8c.keystore`: This file is used to sign the debug build, which is something CI does. We can verify it works locally by applying the following patch:

```diff
--- a/WooCommerce/build.gradle
+++ b/WooCommerce/build.gradle
@@ -387,7 +387,7 @@ if (["uploadStoreFile", "uploadStorePassword", "uploadKeyAlias", "uploadKeyPassw
 
 if (project.hasProperty("debugStoreFile")) {
     def sharedDebugStore = file(project.debugStoreFile.replaceFirst("^~", System.getProperty("user.home")))
-    if (sharedDebugStore.exists()) {
         android {
             signingConfigs {
                 debug {
@@ -395,7 +395,7 @@ if (project.hasProperty("debugStoreFile")) {
                 }
             }
         }
-    }
 }
 
 // Keep this at the bottom (https://stackoverflow.com/a/37565535)
```

After the patch, you should run `./gradlew configureApply` to ensure the gradle.properties local secret is updated. Then remove the keystore with `rm ~/.configure/woocommerce-android/secrets/debug_a8c.keystore`. Finally, run `./gradlew assembleDebug` this should fail because the keystore can't be found and ensure that the repo configuration reads the keystore from the new location. Run `./gradlew configureApply` to restore the secret and `./gradlew assembleDebug` should now pass.

`google-upload-credentials.json`: This file is used by the `upload_to_play_store` Fastlane action. I checked the action [documentation](https://docs.fastlane.tools/actions/upload_to_play_store/) and couldn't find a way to perform a dry-run. I saw there is a `validate_only` option, but the documentation says "Only validate changes with Google Play rather than actually publish," which doesn't make it clear to me whether data would be uploaded or not. I'm not familiar with the upload steps in this repo and would like to avoid uploading stuff accidentally.

@jkmassel let me know if you'd be up to update this PR to be against `release/7.2` to validate the changes sooner and I'll make the switch (or you can do it, if you want).
